### PR TITLE
Robert Rak - Sprint-Challenge-Mongo - Finished

### PR DESCRIPTION
**Pull request by rrak22**

### CODE REVIEW

File: Answers.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Database #
```
Comment (lefrenk) [inline Answers.md]: ✔️

File: budgets/budgetsRouter.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const router = express.Router();
+
+const Budget = require('./Budget.js');
+
+router.route('/')
+  .post(async (req, res) => {
```
Comment (lefrenk) [inline budgets/budgetsRouter.js]: Love it! 👌 `async`/`await`

File: expenses/expensesRouter.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+router.route('/')
+  .get(async (req, res) => {
+    try {
+      const expenses = await Expense.find({}).populate('budget category');
+      res.status(200).json(expenses);
+    } catch (error) {
+      res.status(500).json({ error: "The server has failed you" });
+    }
+  })
+  .post(async (req, res) => {
```
Comment (lefrenk) [inline expenses/expensesRouter.js]: We're missing a key element here for the post function, and that's saving refs to the Expense instance.

Here's an example of saving refs to Budget and Category to an Expense:

```
  .post(async (req, res) => {
    const { amount, budget, category, description } = req.body
    try {
      const cat = await Category.find({ title: category })
      const bud = await Budget.find({ title: budget })

      const expense = new Expense({
        amount,
        budget: bud[0]._id,
        category: cat[0]._id,
        description
      })
      await expense.save()
      res.status(200).json(expense)
    } catch (error) {
      res.status(500).json({ error: error.message })
    }
  })
```

Make sure your POST `req.body` has:
```
{
    "description": "description",
    "amount": 210,
    "budget": "aaa",
    "category": "Food/Dining"
}
```
And that `'aaa'` and `'Food/Dining'` exist in Budget and Category DB, respectively

**Comment by lefrenk** [comment]
Good work, good sir! Really dig that you're implementing `async`/`await` into your routes. Really not much to add here, your code formatting is consistently clean and well-written.

The only thing I would suggest is to practice _saving_ and _populating_ refs. (http://mongoosejs.com/docs/populate.html#saving-refs), and check out the `.select()` method!

File: expenses/expensesRouter.js (hunk 3 above)
Comment (rrak22) [reply expenses/expensesRouter.js]: Great point, totally forgot that! Honestly, I was not visualizing this project well. Definitely should have stepped back and considered the overall picture better.

**Comment by rrak22** [comment]
Thanks man, and I totally agree. I’m going to take your advice and really go over the mongo/mongoose docs in greater detail this weekend.

**Pull request closed without merging**